From 7282aef44c7a00f9914ff1ef73d676483892356f Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Tue, 9 Oct 2018 05:31:00 +0300
Subject: [PATCH 0230/1051] soc: octeontx2: Unset the HWS hardware group mask
 bits

As the LFs will be assigned to different PF/VF combinations the default
mask for having all hwgrps is not a valid case. So unset all of them and
control this via SSOW_LF_GWS_GRPMSK_CHG.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_sso.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_sso.c b/drivers/soc/marvell/octeontx2/rvu_sso.c
index 6af12a02c8d5..b9bc6e611fa4 100644
--- a/drivers/soc/marvell/octeontx2/rvu_sso.c
+++ b/drivers/soc/marvell/octeontx2/rvu_sso.c
@@ -502,7 +502,7 @@ int rvu_sso_init(struct rvu *rvu)
 	u64 iaq_free_cnt, iaq_rsvd, iaq_max, iaq_rsvd_cnt = 0;
 	u64 taq_free_cnt, taq_rsvd, taq_max, taq_rsvd_cnt = 0;
 	struct sso_rsrc *sso = &rvu->hw->sso;
-	int blkaddr, hwgrp, err;
+	int blkaddr, hwgrp, grpmsk, hws, err;
 	u64 reg;
 
 	blkaddr = rvu_get_blkaddr(rvu, BLKTYPE_SSO, 0);
@@ -587,6 +587,19 @@ int rvu_sso_init(struct rvu *rvu)
 			    SSO_AF_TAQ_RSVD_FREE_SHIFT);
 	}
 
+	/* Unset the HWS Hardware Group Mask.
+	 * The hardware group mask should be set by PF/VF
+	 * using SSOW_LF_GWS_GRPMSK_CHG based on the LF allocations.
+	 */
+	for (grpmsk = 0; grpmsk < (sso->sso_hwgrps / 64); grpmsk++) {
+		for (hws = 0; hws < sso->sso_hws; hws++) {
+			rvu_write64(rvu, blkaddr,
+				   SSO_AF_HWSX_SX_GRPMSKX(hws, 0, grpmsk), 0x0);
+			rvu_write64(rvu, blkaddr,
+				   SSO_AF_HWSX_SX_GRPMSKX(hws, 1, grpmsk), 0x0);
+		}
+	}
+
 	/* Allocate SSO_AF_CONST::HWS + 1. As the total number of pf/vf are
 	 * limited by the numeber of HWS available.
 	 */
-- 
2.17.1

